### ISSUE TITLE
monocle looks only for routing updates? 

### ISSUE THREAD

**Issue by shyamkkhadka** [bgpkit/monocle#68]
I could not find a way to look for RIB files. For example, I found all the prefixes that originated from an ASN using `-m, --elem-type <ELEM_TYPE> ` option specifying announcement/withdrawal.

**Comment by digizeph** [comment]
Assuming you're talking about monocle search, it is designed to process only updates files. It mainly looks for "when" some announcements were made and RIB dumps do not provide that information.

You can use "monocle parse" command to handle individual RIB dump files if needed.

**Comment by shyamkkhadka** [comment]
Yes, I asked about the monocle search. Now, I understand the purpose of monocle search. 

It would be nice if `monocle parse` took default argument looking into all MRT files without specifying a single file. 
Anyway, thanks for the information.

**Comment by digizeph** [comment]
Can you help me understand what you mean by "all the RIB dump files"?

"monocle parse" will likely remains parsing only the specified files. It is possible to add option to "monocle search" to forcing it looking at RIB dump files only, but I will need to have some more details on the use case to design an appropriate behaviors for it.

In any case, happy to help

**Comment by shyamkkhadka** [comment]
I meant to look for all the RIB dump files which Routeviews and RIS projects update in certain intervals. As I have been using BGPreader, I think my mind is pre-occupied with its features and expecting the same with the monocle.

**Comment by digizeph** [comment]
I understood now. Thanks for the clarification!

I think we can add the `monocle search --dump-type` to specify the data types:
* `updates` (default): updates files only
* `rib`: rib files only
* `updates,rib`: both files
It'll be a comma-separated string argument.

It will not be a breaking change and allows users to ask for RIB or both if needed. What do you think?

**Comment by shyamkkhadka** [comment]
That is exactly what I thought of.

**Comment by digizeph** [comment]
`--dump-type` filter support added in `v0.7.2`.
